### PR TITLE
Rework GenEd override structure

### PR DESCRIPTION
**Pull request by bombnp**
## Why did you create this PR

- Originally, we planned to store override data based on each semester, academicYear and studyProgram as well to keep genEd mappings with sections. That proved to be way to time-consuming to maintain. We decided to map only courseNo to genEdTypes instead for ease of maintanence.

## What did you do

- Change mongo override schema from
```ts
interface Override {
    courseNo: string
    studyProgram: string
    semester: string
    academicYear: string
    genEd: {
        genEdType: string
        sections: string[]
    }
}
```
to
```ts
interface Override {
    courseNo: string
    genEdType: string
}
```
- update api and reg-scraper to support new override schema
- update codegen
- add new python script to upload overrides

## Demo

[https://dev.cugetreg.com](https://dev.cugetreg.com)

## Checklist

- [x] Deploy a demo
- [ ] Check browsers compatibility
- [ ] Wrote coverage tests

### CODE REVIEW

File: tools/gened/upload_override_gened.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    "beta": "https://beta.cugetreg.com/_api/graphql",
+    "dev": "https://dev.cugetreg.com/_api/graphql",
+    "local": "http://localhost:"+local_port+"/_api/graphql",
+}
+for env in envs:
+    if env not in ["production", "beta", "dev", "local"]:
+        raise Exception("Invalid environment: " + env)
+    print("Uploading overrides to", api_urls[env])
+
+with open(sections_file, mode="r", encoding="utf-8-sig") as csvFile:
```
Comment (saenyakorn) [inline tools/gened/upload_override_gened.py:40]: Please write the comment for the structure of `sections_files` like

```suggestion
# section_files contains X columns as follows:
# courseName: the name of course in Chulalongkorn University
# courseNo: ...
# genEdTyp: ...
with open(sections_file, mode="r", encoding="utf-8-sig") as csvFile:
  # do things
```

Comment (saenyakorn) [reply tools/gened/upload_override_gened.py:40]: Also, wondering why not use `.ipynb` instead? For better data visualization and documentation

**Comment by saenyakorn** [comment]
Not breaking in dev. But, wonder where is 2565 summer

**Comment by bombnp** [comment]
> Not breaking in dev. But, wonder where is 2565 summer

that is weird, it's the web image is not deployed on dev

File: tools/gened/upload_override_gened.py (hunk 1 above)
Comment (bombnp) [reply tools/gened/upload_override_gened.py:40]: idk, just didn't think of it.

Comment (bombnp) [reply tools/gened/upload_override_gened.py:40]: Also this will be deprecated after we have backoffice anyways

**Comment by bombnp** [comment]
> Not breaking in dev. But, wonder where is 2565 summer

it's there now, in dev env

File: tools/gened/upload_override_gened.py (hunk 1 above)
Comment (saenyakorn) [reply tools/gened/upload_override_gened.py:40]: FYI, Jupyter notebook has interactive control feature which allow you to use `input type=select`, `radio` and others. The blog that talking about this feature is here https://towardsdatascience.com/interactive-controls-for-jupyter-notebooks-f5c94829aee6

**Pull request merged**